cli/src/main/java/org/eclipse/hono/cli/AbstractSubCommand.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.cli;
+
+import java.util.concurrent.CountDownLatch;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+
+/**
+ * Abstract base class for the class containing the sub commands execution.
+ */
+public abstract class AbstractSubCommand {
```

Comment:
what is a *SubCommand*? In Spring Shell parlais, these are all just commands, right?
So this might be an *AbstractCommand*?